H3Lib/Extensions/GeoPolygonExtensions.cs

```diff
@@ -186,11 +184,11 @@ public static (int, List<H3Index>) PolyFillInternal(this GeoPolygon polygon, int
                 preSearch.AddRange(innerHex);
             }
 
-            search = new HashSet<H3Index>(numHexagons);
+            search = new HashSet<H3Index>();
             search.UnionWith(preSearch);
             
             var found = new HashSet<H3Index>();
-            var results = new HashSet<H3Index>(numHexagons);
+            var results = new HashSet<H3Index>();
```

Comment:
This may be good, though I think I put that there initially to raise an exception if attempts were made to put in more than `numHexagons` values.  I'll go through the code later after work.